rlclientlib/serialization/payload_serializer.h

```diff
@@ -0,0 +1,134 @@
+#pragma once
+
+#pragma once
+
+#include <vector>
+
+#include <flatbuffers/flatbuffers.h>
+
+#include "action_flags.h"
+#include "ranking_event.h"
+#include "data_buffer.h"
+#include "logger/message_type.h"
+#include "api_status.h"
+#include "utility/data_buffer_streambuf.h"
+#include "learning_mode.h"
+
+#include "generated/v2/OutcomeSingle_generated.h"
+#include "generated/v2/CbEvent_generated.h"
+#include "generated/v2/CcbEvent_generated.h"
+#include "generated/v2/SlatesEvent_generated.h"
+
+namespace reinforcement_learning {
+  namespace logger {
+    using namespace messages::flatbuff;
+
+    v2::LearningModeType GetLearningMode(learning_mode mode);
+
+    template<payload_type pt>
+    struct payload_serializer {
+      const payload_type type = pt;
+    };
+
+    struct cb_serializer : payload_serializer<payload_type::CB> {
+      static generic_event::payload_buffer_t event(const char* context, unsigned int flags, learning_mode learning_mode, const ranking_response& response) {
+        flatbuffers::FlatBufferBuilder fbb;
+        std::vector<uint64_t> action_ids;
+        std::vector<float> probabilities;
+        for (auto const& r : response) {
+          action_ids.push_back(r.action_id + 1);
+          probabilities.push_back(r.probability);
+        }
+        std::vector<unsigned char> _context;
+        std::string context_str(context);
+        copy(context_str.begin(), context_str.end(), std::back_inserter(_context));
+
+        auto fb = v2::CreateCbEventDirect(fbb, flags | action_flags::DEFERRED, &action_ids, &_context, &probabilities, response.get_model_id(), GetLearningMode(learning_mode));
```

Comment:
setting action_flags::DEFERRED here looks wrong.

Reply:
good catch, thank you